services/alertmanager/alertmanager.go

```diff
@@ -328,6 +332,97 @@ func (svc *Service) configAndReload(ctx context.Context, b []byte) error {
 	return nil
 }
 
+func convertTLSConfig(id string, tls *models.TLSConfig) promconfig.TLSConfig {
+	hashedIDBytes := sha256.Sum256([]byte(id))
+	hashedID := hex.EncodeToString(hashedIDBytes[:])
+
+	caFile := tls.CAFile
+	if tls.CAFileContent != "" {
+		caFile = path.Join(alertmanagerCertDir, fmt.Sprintf("%s.ca", hashedID))
+	}
+	certFile := tls.CertFile
+	if tls.CertFileContent != "" {
+		certFile = path.Join(alertmanagerCertDir, fmt.Sprintf("%s.crt", hashedID))
+	}
+	keyFile := tls.KeyFile
+	if tls.KeyFileContent != "" {
+		keyFile = path.Join(alertmanagerCertDir, fmt.Sprintf("%s.key", hashedID))
+	}
+	return promconfig.TLSConfig{
+		CAFile:             caFile,
+		CertFile:           certFile,
+		KeyFile:            keyFile,
+		ServerName:         tls.ServerName,
+		InsecureSkipVerify: tls.InsecureSkipVerify,
+	}
+}
+
+// recreateTLSConfigFiles cleanups old tls config files and creates new ones using content from the DB.
```

Comment:
this comment is little misleading, because you don't access DB

Reply:
reworked